### PR TITLE
[Testing] Adding smoke test for gcc ABI

### PR DESCRIPTION
**Pull request by pjh5**
Not ready yet.

### CODE REVIEW

File: check_binary.sh [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  if [[ "$(is_expected "$actual_gcc_abi")" != 1 ]]; then
+    echo "gcc ABI $actual_gcc_abi not as expected."
+    exit 1
+  fi
+
+  # We also check that there are[not] cxx11 symbols in libtorch
+  echo "Checking that symbols in libcaffe2.so have the right gcc abi"
+  libtorch="${install_root}/lib/libcaffe2.so"
+  set +e
+  cxx11_symbols="$(nm "$libtorch" | c++filt | grep cxx11)"
```
Comment (bwasti) [inline check_binary.sh]: grep for `__cxx11` rather than `cxx11`

File: check_binary.sh [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  has_cxx11_symbols="$?"
+  set -e
+  if [[ "$(is_expected "$has_cxx11_symbols")" != 1 ]]; then
+    if [[ "$has_cxx11_symbols" == 0 ]]; then
```
Comment (bwasti) [inline check_binary.sh]: I think this is backwards. Doesn't `$?` give the return code, which would be 0 when symbols are found

File: check_binary.sh [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+  #   quote, any characters
+  #   Note the exactly one single character after the '='. In the case that the
+  #     variable is not set the '=' will be followed by a '"' immediately and the
+  #     line will fail the match and nothing will be printed; this is what we
+  #     want.  Otherwise it will capture the 0 or 1 after the '='.
+  # /.*D_GLIBCXX_USE_CXX11_ABI=\(.\)".*/
+  #   replace the matched line with the capture group and print
+  # /\1/p
+  actual_gcc_abi="$(sed -ne 's/.*D_GLIBCXX_USE_CXX11_ABI=\(.\)".*/\1/p' < "$torch_config")"
+  if [[ "$(is_expected "$actual_gcc_abi")" != 1 ]]; then
```
Comment (bwasti) [inline check_binary.sh]: I'm not a bash expert, but in general I think it is best practice to use 
```
"$(is_expected ${actual_gcc_abi})"
```

File: check_binary.sh [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+python -c 'import torch'
+python -c 'from caffe2.python import core'
+
+
+###############################################################################
+# Check for MKL
+###############################################################################
+if [[ "$(uname)" != 'Darwin' || "$package_type" != *wheel ]]; then
+    echo "Checking that MKL is available"
+    python -c 'import torch; exit(0 if torch.backends.mkl.is_available() else 1)'
```
Comment (bwasti) [inline check_binary.sh]: `set -e`?

File: check_binary.sh [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+###############################################################################
+# Check CUDA configured correctly
+###############################################################################
+# Skip these for Windows machines without GPUs
+if [[ "$OSTYPE" == "msys" ]]; then
+    GPUS=$(wmic path win32_VideoController get name)
+    if [[ ! "$GPUS" == *NVIDIA* ]]; then
+        echo "Skip CUDA tests for machines without a Nvidia GPU card"
+        exit 0
```
Comment (bwasti) [inline check_binary.sh]: perhaps just set `DESIRED_CUDA='cpu'`?

File: check_binary.sh (hunk 2 above)
Comment (pjh5) [reply check_binary.sh]: Yes :(

File: check_binary.sh (hunk 5 above)
Comment (pjh5) [reply check_binary.sh]: @peterjc123 has context on this. I only copied this from another file

**Comment by pjh5** [comment]
Judging from https://github.com/pytorch/pytorch/pull/20507 , this actually looks safe to land now before the associated pytorch/pytorch changes

**Pull request merged**